packages/shared/src/types.ts

```diff
@@ -1,3 +1,12 @@
+import { ethers } from 'ethers';
+
+import { TacoSigner } from './taco-interfaces';
+import { Account, PublicClient } from './viem/types';
+
 export type ChecksumAddress = `0x${string}`;
 export type HexEncodedBytes = string;
 export type Base64EncodedBytes = string;
+
+export type ProviderLike = ethers.providers.Provider | PublicClient;
+
+export type SignerLike = TacoSigner | Account;
```

Comment:
If this isn't the following:

```suggestion
export type SignerLike = ethers.Signer | Account;
```

Doesn't that affect our API i.e. anything that took an `ethers.Signer` before and now takes `SignerLike`, basically no longer accepts an `ethers.Signer` anymore?

`TacoSigner` is purely for internal usage, and not external facing, no?

Reply:
Thanks @derekpierre for the question,
`TacoSigner` has a subset of `ethers.Signer` functions. For that, TypeScript compiler, for any variable typed `TacoSigner`, allows it to be assigned an instance of `ethers.Signer`. In other words every `ethers.Signer` implicitly implements the interface `TacoSigner`. i.e. the TypeScript compiler handles `ethers.Signer` as if it was `class ethers.Signer implements TacoSigner {...}` even though this is not explicitly written.

And this means for any function that was accepting `ethers.Signer`, and now accepting `TacoSigner`, it still has 100% backward compatibility.

And I just updated some of the JSDocs to hint that.